src/liballoc/raw_vec.rs

```diff
@@ -131,17 +142,30 @@ impl<T> RawVec<T, Global> {
     /// `RawVec` with capacity `0`. If `T` is zero-sized, then it makes a
     /// `RawVec` with capacity `usize::MAX`. Useful for implementing
     /// delayed allocation.
+    #[cfg(not(bootstrap))]
     pub const fn new() -> Self {
         // FIXME(Centril): Reintegrate this with `fn new_in` when we can.
```

Comment:
Can I do this in this PR? Or should I leave it?

Reply:
I would say do it in this PR. @Centril?

Reply:
I did it, but let's wait for @Centril 

Also, I left the hack above that (`RawVec::NEW`) because removing it causes errors where `Vec:new` complains for some reason...

Reply:
If you change this now then it wouldn't be possible to implement the `RawVec::NEW` comment because `const_if_match` isn't stable yet. That said, I think we can change the implementation now and change it back if we need to satisfy the `HACK` comment.

Reply:
@Centril The CI seems to pass with `RawVec::NEW = Self::new()`?

Perhaps you could explain what exactly the hack is and why it is needed? I don't really understand it's purpose.

Reply:
@mark-i-m The hack is that `RawVec::NEW` exists to begin with. We are abusing the fact that we don't look into constants to enforce `min_const_fn` so you can use unstable const features in it. We had to do this because the instability of `Self::new` means that it's checked and recorded as a `min_const_fn` and therefore cannot be called as one. What we would need is to separate things so that unstable things can be checked as `min_const_fn` and callable as such as well. See https://github.com/rust-lang/rust/issues/64285.

Reply:
So if I understand you correctly, it's not a safety thing but a matter of getting it to compile without requiring users to enable the feature (since it needs to be usable on stable). Is that correct?

Reply:
It's more about internal libstd checks and ensuring that we don't accidentally allow unstable const stuff in stable `const fn`s transitively.